akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java

```diff
@@ -311,7 +311,9 @@ public Receive createReceive() {
   //#ask-actor
 
   //#actorRefWithAck-actor
-  static class Ack {}
+  enum Ack {
+    INSTANCE;
+  }
```

Comment:
How do we usually encode singletons in Java examples? I looked around a bit but only found `static class Foo {}` - which is not the same, since for those different instances don't `.equals` each other.

Reply:
That pattern is fine IMO.

"Usually" we'd do classes and a getInstance is what Johan means I guess. Maintaining the equals manually is a pain tho. I'm ok with the enum here 👍

Reply:
Most of the time when we write these we write them in Scala and `case object` + `getInstance` is trivial then; in java nope though

Reply:
👍 then